### PR TITLE
feat: initial files

### PR DESCRIPTION
**Pull request by lucasgrvarela**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  depends_on = [
+    google_project_service.service,
+    google_project_iam_member.service-account,
+    google_project_iam_member.service-account-custom,
+    google_compute_router_nat.nat,
+  ]
+
+}
+
+// A dedicated/separate node pool where workloads will run.  A regional node pool
```
Comment (EvertonSA) [inline main.tf]: remove comments

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+// will have "node_count" nodes per zone, and will use 3 zones.  This node pool
+// will be 3 nodes in size and use a non-default service-account with minimal
+// Oauth scope permissions.
+resource "google_container_node_pool" "private-np-1" {
+  provider = google-beta
+
+  name       = "private-np-1"
```
Comment (EvertonSA) [inline main.tf]: parameterize node pool name using variables.tf

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  node_count = "3"
+
+  // Repair any issues but don't auto upgrade node versions
+  management {
+    auto_repair  = "true"
+    auto_upgrade = "false"
+  }
+
+  node_config {
+    machine_type = "n1-standard-2"
```
Comment (EvertonSA) [inline main.tf]: parameterize machine_type using variables.tf

File: network.tf [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+  sudo apt-get update -y
+  sudo apt-get install -y tinyproxy
+  EOF
+
+}
+
+// The Bastion Host
+resource "google_compute_instance" "bastion" {
+  name = local.hostname
+  machine_type = "g1-small"
```
Comment (EvertonSA) [inline network.tf]: no need to parametrize, but please use a smaller instance to avoid paying

File: network.tf [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+  zone = format("%s-a", var.region)
+  project = var.project
+  tags = ["bastion"]
+
+  // Specify the Operating System Family and version.
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-9"
```
Comment (EvertonSA) [inline network.tf]: is possible please use centos

**Comment by EvertonSA** [comment]
Please use the following folder structure:

dev/
prd/ <- empty folder
init.sh 
README.md

Readme should include how to access the cluster using the bastion server, how to provision the cluster from scratch. Can be in portuguese.

init.sh should:
activate gcloud services.
provision buckets (dev and prd) for storing terraform state. 

Also please change the terraform to use the backend bucket provisioned in the init.sh

**Review by EvertonSA — COMMENTED** [review]
vc pode ver esses pontos?

File: main.tf (hunk 3 above)
Comment (lucasgrvarela) [reply main.tf]: do you want the machine_type to be started with n1-standard-1 or n1-standard-2?

Comment (EvertonSA) [reply main.tf]: 2 x n1-highmem-2

**Comment by lucasgrvarela** [comment]
> Please use the following folder structure:
> 
> dev/
> prd/ <- empty folder
> init.sh
> README.md
> 
> Readme should include how to access the cluster using the bastion server, how to provision the cluster from scratch. Can be in portuguese.
> 
> init.sh should:
> activate gcloud services.
> provision buckets (dev and prd) for storing terraform state.
> 
> Also please change the terraform to use the backend bucket provisioned in the init.sh

Done!

**Pull request merged**